### PR TITLE
[REG-1477] Add PostHog analytics to docs site

### PR DESCRIPTION
**Pull request by vontell**
Adds PostHog analytics to our docs site. This will now allow us to determine where visitors are going when they leave our site to go to the docs site. Verified that this is working because I can see events coming in within posthog.

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [x] The code is extensible and backward compatible
- [x] New public interfaces are extensible and open to backward compatibility in the future
- [x] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [x] Non-critical or potentially modifiable arguments are optional
- [x] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [x] The code is easy to read
- [x] Unit tests are added for expected and edge cases
- [x] Integration tests are added for expected and edge cases
- [x] Functions and classes are documented
- [x] Migrations for both up and down operations are completed
- [x] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [x] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [x] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -148,6 +148,13 @@ const config = {
         additionalLanguages: ['csharp'],
       },
     }),
+    scripts: [
+      {
+        src:
+          '/js/loadposthog.js',
+        async: false,
```
Comment (RG-nAmKcAz) [inline docusaurus.config.js:155]: i'm assuming that blocking the rendering while this script loads is intentional ?

**Review by RG-nAmKcAz — APPROVED** [review]

**Review by addisonbgross — APPROVED** [review]
Approved pending @RG-nAmKcAz 's comment

File: docusaurus.config.js (hunk 1 above)
Comment (vontell) [reply docusaurus.config.js:155]: Yup! This is what is recommended for proper logging of events.

**Review by analogrelay — APPROVED** [review]
